crates/wasi-common/src/ctx.rs

```diff
@@ -118,11 +119,28 @@ impl WasiCtx {
             .push(Arc::new(DirEntry::new(Some(path.as_ref().to_owned()), dir)))?;
         Ok(())
     }
+
+    // handle how the wasi context should be cloned when fork happens
+    pub fn fork(&self) -> Self {
+        let forked_ctx = WasiCtxInner {
+            args: self.args.clone(), // we want to clone the entire args
+            env: self.env.clone(), // as well as environment variables
+            // below are currently not used by glibc in lind-wasm, so doesn't really matter how to handle them for now
+            random: Mutex::new(random_ctx()),
```

Comment:
I don't really understand the following args but I assumed they are refered to in the original CTX struct. Any notes here are helpful though.